File: apps/party-time-backend/src/main/java/com/partytime/jpa/repository/AccountRepository.java
package com.partytime.jpa.repository;

import com.partytime.jpa.entity.Account;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface AccountRepository extends JpaRepository<Account, Long> {

    Optional<Account> findByEmailVerificationCode(String code);

    boolean existsByEmail(String mail);

    Optional<Account> findAccountByEmail(String mail);

}

File: apps/party-time-backend-kotlin/src/main/kotlin/com/partytime/jpa/repository/AccountRepository.kt
package com.partytime.jpa.repository

import com.partytime.jpa.entity.Account
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface AccountRepository: JpaRepository<Account, Long> {
    fun findByEmailVerificationCode(code: String): Optional<Account>
    fun existsByEmail(mail: String): Boolean
    fun findAccountByEmail(mail: String): Optional<Account>
}
